### PR TITLE
load all substitutions from etc

### PR DESCRIPTION
**Pull request by evgeni**
On some distributions (like CentOS Stream and Oracle Linux), we need more substitutions to be able to load repositories properly.

DNF has a helper for that: conf.substitutions.update_from_etc.

On pure DNF distributions, calling this should be sufficient. On EL7, where the primary tool is YUM, DNF does not load vars from /etc/yum, only from /etc/dnf, so we have to help it a bit and explicitly try to load releasever from /etc/yum.
(DNF since 4.2.15 *does* also load substitutions from /etc/yum, but EL7 ships with 4.0.x)

### CODE REVIEW

**Comment by evgeni** [comment]
review please @oamg/developers

File: repos/system_upgrade/common/libraries/module.py [hunk 1]
```diff
@@ -23,14 +23,16 @@ def _create_or_get_dnf_base(base=None):
...
         # /etc/yum/vars/releasever is present, read its contents so that we can access repositores on such systems.
         conf = dnf.conf.Conf()
-        pkg_manager = 'yum' if get_source_major_version() == '7' else 'dnf'
-        releasever_path = '/etc/{0}/vars/releasever'.format(pkg_manager)
-        if os.path.exists(releasever_path):
+
+        # dnf on EL7 doesn't load vars from /etc/yum, so we need to help it a bit
+        releasever_path = '/etc/yum/vars/releasever'
+        if get_source_major_version() == '7' and os.path.exists(releasever_path):
             with open(releasever_path) as releasever_file:
```
Comment (abadger) [inline repos/system_upgrade/common/libraries/module.py:30]: I would put the releasever_path setting inside of the condition to show that it is only a part of the code needed to handle EL7:
```suggestion
        # dnf on EL7 doesn't load vars from /etc/yum, so we need to help it a bit
        if get_source_major_version() == '7' and os.path.exists(releasever_path):
            releasever_path = '/etc/yum/vars/releasever'
            with open(releasever_path) as releasever_file:
```

A general note: using a `try: except:` around `open()` and passing if the file does not exist is resistant to race conditions whereas checking file existence first and then operating on the file is not so I generally prefer the `try: except:` strategy.  However, I see (1) that we use the file existence check strategy all over the code right now and (2) this specific case is unlikely to have the file deleted out from under us. So, something to keep in mind but I won't require that change to be made.

**Review by abadger — APPROVED** [review]
I don't know the dnf and yum apis intimately but the reasoning here makes sense and I've looked at the relevant code in `dnf.conf.substitutions.update_from_etc()` to check that it takes care of this case in EL8.

The only changes I asked for are cosmetic so I'm approving this.  You can add my suggestion or let me know you'd rather not and then this can be merged.

File: repos/system_upgrade/common/libraries/module.py (hunk 1 above)
Comment (evgeni) [reply repos/system_upgrade/common/libraries/module.py:30]: I can't call `os.path.exists(releasever_path)` before assigning `releasever_path = '/etc/yum/vars/releasever'` ;-)

The logic I wanted to keep is "if on EL7 *and* the file exists, load it, in *all* other cases use `get_source_major_version`.

I could rewrite this to:
```
conf.substitutions['releasever'] = get_source_major_version()
if get_source_major_version() == '7':
  releasever_path = '/etc/yum/vars/releasever'
  if os.path.exists(releasever_path):
    with …:
      conf.substitutions['releasever'] = releasever_file.read().strip()
```

But it doesn't make the code much more readable, IMHO.

Comment (evgeni) [reply repos/system_upgrade/common/libraries/module.py:30]: Or even
```python
conf.substitutions['releasever'] = get_source_major_version()

if get_source_major_version() == '7':
  try:
    with open('/etc/yum/vars/releasever') as releasever_file:
      conf.substitutions['releasever'] = releasever_file.read().strip()
  except:
    pass

# load all substitutions from etc
conf.substitutions.update_from_etc('/')
```

Comment (evgeni) [reply repos/system_upgrade/common/libraries/module.py]: Done this now. Let me know what you think.

Comment (abadger) [reply repos/system_upgrade/common/libraries/module.py]: +1

Comment (abadger) [reply repos/system_upgrade/common/libraries/module.py]: And yeah, I got ahead of myself thinking about both changing to `try: except:` and moving the assignment but then only making one of those a suggestion.

**Review by abadger — APPROVED** [review]
Looks good

**Comment by abadger** [comment]
Thanks @evgeni !

**Pull request merged**